src/v/security/audit/schemas/schemas.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "types.h"
+
+namespace security::audit {
+
+template<typename T>
+ss::sstring rjson_serialize(const T& v) {
+    ::json::StringBuffer str_buf;
+    ::json::Writer<::json::StringBuffer> wrt(str_buf);
+
+    using ::json::rjson_serialize;
+    using ::security::audit::rjson_serialize;
+
+    rjson_serialize(wrt, v);
+
+    return ss::sstring{str_buf.GetString(), str_buf.GetSize()};
+}
+
+class ocsf_base_event {
+public:
+    auto equality_fields() const {
+        return std::tie(
+          category_uid, class_uid, metadata, severity_id, type_uid);
+    }
+
+    void increment(timestamp_t event_time) const {
+        this->count++;
+        this->end_time = event_time;
+    }
+
+    size_t key() const noexcept {
```

Comment:
Not understanding this, looks like subclasses just set `_key` to `hash()`, so why can't `hash()` be public? and we can remove `key` and the method `key()` as well

Reply:
This is so we don't waste cycles and regenerate the `hash()` every time we need the key/hash (such as when inserting into a map or searching the map).

Reply:
I think its fine here, usually the trade off is extra memory anyway